lib/dogapi/v1/service_level_objective.rb

```diff
@@ -0,0 +1,113 @@
+require 'dogapi'
+
+module Dogapi
+  class V1 # for namespacing
+
+    # Implements Service Level Objectives endpoints
+    class ServiceLevelObjectiveService < Dogapi::APIService
+
+      API_VERSION = 'v1'
+
+      def create_service_level_objective(type, name, description = nil, tags = nil, thresholds = nil, numerator = nil,
+                                         denominator = nil, monitor_ids = nil, monitor_search = nil, groups = nil)
```

Comment:
Looks like you want to have keywords arguments here. Otherwise, if you want to specify monitor SLO, you'll have to explicitly set `denominator` and `numerator` to `nil`, which is not great.
Another option, which some endpoints already use, is to accept a single `options` dict argument for everything that is optional (like the monitor endpoint: https://github.com/datadog/dogapi-rb/blob/07358ed8e1c7e7399149f5972244fca90d5a73da/lib/dogapi/v1/monitor.rb#L10)

Reply:
`options` in monitors is actually referencing the `monitor.options` stanza which is dynamic. 
in this case all these parameters are top-level arguments to the SLO. 

the alternative is to use the named method caller: `create_service_level_objective('monitor', 'foo', monitor_ids: [1, 2, 3])` the rest of the arguments default to their `default` argument value.

Reply:
Sounds good, but right now you can't give named arguments. You need something like this instead 
```ruby
def create_service_level_objective(type, name, description: nil, tags: nil, thresholds: nil, numerator: nil,
                                          denominator: nil, monitor_ids: nil, monitor_search: nil, groups: nil)
```

Reply:
oops, updating :)

Reply:
Thanks ! Now let's update the test to have the proper calls for the functions

Reply:
yes, working on that now actually. the shared helper tests use `*args` so I'm about to push a fix with an alternative with named args `**args`